Dockerfile.extra

```diff
@@ -0,0 +1,58 @@
+# -- plugins -------------------------------------------------------------------
+
+FROM development AS plugins
+
+WORKDIR /tmp/vast
+
+RUN cmake -S contrib/vast-plugins/compaction -B build-compaction -G Ninja \
+      -D CMAKE_INSTALL_PREFIX:STRING="$PREFIX" && \
+      cmake --build build-compaction --parallel && \
+      cmake --install build-compaction --strip && \
+      rm -rf build-compaction
+
+RUN cmake -S contrib/vast-plugins/fleet -B build-fleet -G Ninja \
+      -D CMAKE_INSTALL_PREFIX:STRING="$PREFIX" && \
+      cmake --build build-fleet --parallel && \
+      cmake --install build-fleet --strip && \
+      rm -rf build-fleet
+
+RUN cmake -S contrib/vast-plugins/inventory -B build-inventory -G Ninja \
+      -D CMAKE_INSTALL_PREFIX:STRING="$PREFIX" && \
+      cmake --build build-inventory --parallel && \
+      cmake --install build-inventory --strip && \
+      rm -rf build-inventory
+
+RUN cmake -S contrib/vast-plugins/matcher -B build-matcher -G Ninja \
+      -D CMAKE_INSTALL_PREFIX:STRING="$PREFIX" && \
+      cmake --build build-matcher --parallel && \
+      cmake --install build-matcher --strip && \
+      rm -rf build-matcher
+
+RUN cmake -S contrib/vast-plugins/netflow -B build-netflow -G Ninja \
+      -D CMAKE_INSTALL_PREFIX:STRING="$PREFIX" && \
+      cmake --build build-netflow --parallel && \
+      cmake --install build-netflow --strip && \
+      rm -rf build-netflow
+
+# -- vast-ce -------------------------------------------------------------------
+
+FROM production AS vast-ce
+
+COPY --from=plugins --chown=vast:vast $PREFIX/lib/vast/plugins/libvast-plugin-matcher.so $PREFIX/lib/vast/plugins/
+COPY --from=plugins --chown=vast:vast $PREFIX/share/doc/matcher $PREFIX/share/doc/matcher
```

Comment:
I find this to be incredibly brittle. Can we instead of listing all installed files manually here install the proprietary plugins into a separate prefix and then merge them into VAST's install prefix afterwards?

Reply:
Are we certain that none of the plugins makes use of the install prefix in any way? Maybe not now, but maybe in the future?
I thought about 2 ways to do this cleanly:
1. use the install manifest to generate a tarball that can be copied and extracted in the new image.
2. Use a dedicated build context per plugin and move the existing stuff in the prefix out of the way before installing.
3. Use the install manifest to copy all installed files of a plugin into a dedicated tree and merge it back with the `COPY` directive.

I think I like (3) best, because it has the least overhead while not installing a nasty trap for our future selves.

Reply:
I forgot about `DESTDIR`, which should make this trivial.